applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_coupled_solver.py

```diff
@@ -83,9 +83,18 @@ def Finalize(self):
             coupling_operation.Finalize()
 
     def AdvanceInTime(self, current_time):
+        # not all solvers provide time (e.g. external solvers or steady solvers)
+        # hence we have to check first if they return time (i.e. time != 0.0)
+        # and then if the times are matching, since currently no interpolation in time is possible
+
         self.time = 0.0
         for solver in self.solver_wrappers.values():
-            self.time = max(self.time, solver.AdvanceInTime(current_time))
+            solver_time = solver.AdvanceInTime(current_time)
+            if solver_time != 0.0: # solver provides time
+                if self.time == 0.0: # first time a solver returns a time different from 0.0
+                    self.time = solver_time
+                elif abs(self.time - solver_time) > 1e-12:
```

Comment:
shall we make 1e-12 a class member variable and use it here ?

Reply:
What would be the use?
I think we could do this if we need it, not overengineer it (even more ;) )

Reply:
I suggested because its a hard coded number and i remember reading in some guidelines thatit is discouraged.

Reply:
hm not sure, making it a member makes it modifyable, accessable etc which I don't want. So unless there is a good reason I would leave it for now, we could still change it if we need to in the future